### PR TITLE
Fix DeprecationWarning when using lz4

### PR DESCRIPTION
**Pull request by vetal4444**
When use `python-driver` with latest `lz4` I got a lot of `DeprecationWarning`:

    /usr/local/lib/python3.5/dist-packages/lz4/__init__.py:18: DeprecationWarning: Call to deprecated function or method compress (use lz4.block.compress instead).
      def compress(*args, **kwargs):
    /usr/local/lib/python3.5/dist-packages/lz4/__init__.py:22: DeprecationWarning: Call to deprecated function or method decompress (use lz4.block.decompress instead).
      def decompress(*args, **kwargs):

### CODE REVIEW

**Comment by mambocab** [comment]
Thanks for the PR! I've created [PYTHON-769](https://datastax-oss.atlassian.net/browse/PYTHON-769) to review it.

Just a heads-up, it may take some time to complete this review -- our queue for our next release is already fairly full, and we may have to do a little research on what versions of `lz4` are available on different Linux distros.

**Comment by vetal4444** [comment]
I can change my PR to support old versions on lz4. Something like that:

    lz4_block = lz4.block if hasattr(lz4, "block") else lz4

    def lz4_compress(byts):
            # write length in big-endian instead of little-endian
            return int32_pack(len(byts)) + lz4_block.compress(byts)[4:]

        def lz4_decompress(byts):
            # flip from big-endian to little-endian
            return lz4_block.decompress(byts[3::-1] + byts[4:])

    locally_supported_compressions['lz4'] = (lz4_compress, lz4_decompress)

What do you think about this?

**Comment by mambocab** [comment]
Ah, sure, that'd be reasonable with some documentation. @aboudreault Do you see any issues with the proposed solution?

**Comment by aboudreault** [comment]
I think I would do more a try/except here since lz4.block will work mostly everytime in new envs.
```
try:
    lz4_block = lz4.block
except AttributeError:
    lz4_block = lz4
```

LGTM when the PR is fixed.

File: cassandra/connection.py [hunk 1]
```diff
@@ -61,18 +61,22 @@
 except ImportError:
     pass
 else:
+    # Old versions of lz4 has not block attribute.
+    # In new versions need to use lz4.block, because functions from
+    # lz4 namespace are deprecated.
+    lz4_block = lz4.block if hasattr(lz4, "block") else lz4
```
Comment (aboudreault) [inline cassandra/connection.py]: @vetal4444 Can you make the change I mentioned in the PR conversation about this? Otherwise, LGTM.

**Comment by mambocab** [comment]
Awesome, thank you! I'm going to rebase and merge. Thanks for catching this and working with us to fix it.

**Comment by msenol86** [comment]
I am still getting this error on Ubuntu 14.04, Python 2.7 and cassandra-driver (3.10). I have installed cassandra-driver via pip

**Comment by vetal4444** [comment]
> I am still getting this warning on Ubuntu 14.04, Python 2.7 and cassandra-driver (3.10). I have installed cassandra-driver via pip

Thats because this fix will be included in next release. See [changelog](https://github.com/datastax/python-driver/blob/master/CHANGELOG.rst) for more details.

**Comment by kmatt** [comment]
I'm still seeing these warnings with cassandra-driver 3.12.0 and Python lz4 0.12.0 (on CentOS 7.7.4.1708). This is with cqlsh from the apache-cassandra-3.11.1-bin.tar.gz tarball.

**Comment by bjmb** [comment]
That tarball is bundled with the driver version 3.10, not 3.11, it's probably using the bundled version instead of the one you have installed.

**Comment by kmatt** [comment]
Even with RPM install. Adding `print cassandra.__file__` to cqlsh.py shows me `/usr/share/cassandra/lib/cassandra-driver-internal-only-3.10.zip/cassandra-driver-3.10/cassandra/__init__.py`

Moving cassandra-driver-internal-only-3.10.zip out of the installed path addresses the issue. Where does this driver locate the bundled version of the driver? An option to use one in the Python site-packages path would be handy.

**Pull request merged**